app/controllers/adoption_applications_controller.rb

```diff
@@ -4,17 +4,18 @@ def new
   end
 
   def create
-    @selected_pet_ids = params[:pet_ids]
     @adoption_application = AdoptionApplication.create(adoption_application_params)
     if @adoption_application.save
+      @selected_pet_ids = params[:pet_ids]
       @selected_pet_ids.each do |pet_id|
         session[:favorites].delete(pet_id)
         PetAdoptionApplication.create(adoption_application_id: @adoption_application.id, pet_id: pet_id)
       end
       flash[:notice] = "Your adoption application has been submitted!"
       redirect_to '/favorites'
-    # else
-    #   flash[:notice] = "Unsuccessful application submission: please fill in all application fields."
+    else
+      flash[:notice] = "Unsuccessful application submission: please fill in all application fields."
+      redirect_to "/adoption_applications/new"
```

Comment:
for some reason, an error was thrown when I had this as `render :new` and only worked when I changed it to `redirect_to ...`
if we don't have any bigger questions to ask during our upcoming project checkin, maybe we can get clarity on why it didn't work with the `render`